nano/node/bootstrap/block_deserializer.cpp

```diff
@@ -11,48 +11,56 @@ nano::bootstrap::block_deserializer::block_deserializer () :
 
 }
 
-void nano::bootstrap::block_deserializer::read (std::shared_ptr<nano::socket> socket)
+void nano::bootstrap::block_deserializer::read (std::shared_ptr<nano::socket> socket, callback_type callback)
 {
+	debug_assert (socket != nullptr);
```

Comment:
if we had the reference `nano::socket & socket` we won't need to assert `socket != nullptr`. The same works for the `callback`.